File: app/src/main/java/org/nervos/neuron/view/dialog/SelectorDialog.java
package org.nervos.neuron.view.dialog;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetDialog;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import org.nervos.neuron.R;
import org.nervos.neuron.view.button.FillButton;

/**
 * Created by BaojunCZ on 2018/9/11.
 */
public class SelectorDialog {

    private TextView titleText;
    private ImageView closeImg;
    private RecyclerView recyclerView;
    private FillButton okBtn;
    private Context context;
    private BottomSheetDialog dialog;
    private View view;

    public SelectorDialog(@NonNull Context context) {
        this.context = context;
        dialog = new BottomSheetDialog(context);
        view = LayoutInflater.from(context).inflate(R.layout.view_bottom_selector, null);
        dialog.setContentView(view);
        dialog.show();
        initView();
        initData();
        initAction();
    }

    private void initView() {
        titleText = view.findViewById(R.id.tv_title);
        closeImg = view.findViewById(R.id.iv_close);
        recyclerView = view.findViewById(R.id.recycler);
        okBtn = view.findViewById(R.id.btn_ok);
    }

    private void initData() {

    }

    private void initAction() {
        closeImg.setOnClickListener(view -> dialog.dismiss());
    }

    public void setTitleText(String title) {
        titleText.setText(title);
    }

    public void setOkBtn(String text) {
        okBtn.setText(text);
    }

    public void setOkListner(View.OnClickListener listner) {
        okBtn.setOnClickListener(listner);
    }

    public void setRecyclerView(RecyclerView.Adapter adapter) {
        recyclerView.setLayoutManager(new LinearLayoutManager(context));
        recyclerView.setAdapter(adapter);
    }

    public void dismiss() {
        dialog.dismiss();
    }

}

File: app/src/main/java/com/cryptape/cita_wallet/view/dialog/SelectorDialog.kt
package com.cryptape.cita_wallet.view.dialog

import android.content.Context
import android.content.DialogInterface
import android.support.design.widget.BottomSheetDialog
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.cryptape.cita_wallet.R
import com.cryptape.cita_wallet.util.DipUtils


/**
 * Created by BaojunCZ on 2018/9/11.
 */
open class SelectorDialog(private val context: Context) {

    private val dialog: BottomSheetDialog = BottomSheetDialog(context)
    private val view: View = LayoutInflater.from(context).inflate(R.layout.view_bottom_selector, null)
    private var closeListener: View.OnClickListener? = null
    private var okListener: View.OnClickListener? = null
    private var mRecyclerView = view.findViewById<RecyclerView>(R.id.recycler)
    private var mTvCancel = view.findViewById<TextView>(R.id.tv_cancel)
    private var mTvOk = view.findViewById<TextView>(R.id.tv_ok)

    init {
        dialog.setContentView(view)
        dialog.show()
        initAction()
    }

    private fun initAction() {
        mTvCancel.setOnClickListener { view ->
            if (closeListener != null) {
                closeListener!!.onClick(view)
            } else {
                dialog.dismiss()
            }
        }
        mTvOk.setOnClickListener {
            if (okListener != null) {
                okListener!!.onClick(view)
            } else {
                dialog.dismiss()
            }
        }
    }

    fun setOnCloseListener(listener: View.OnClickListener) {
        this.closeListener = listener
    }

    fun setOnOkListener(listener: View.OnClickListener) {
        this.okListener = listener
    }

    fun setRecyclerView(adapter: RecyclerView.Adapter<*>) {
        mRecyclerView.layoutManager = LinearLayoutManager(context)
        mRecyclerView.adapter = adapter
    }

    fun setRecyclerHeight(height: Float) {
        mRecyclerView.layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, DipUtils.dip2px(context, height))
    }

    fun setOnDismissListener(listener: DialogInterface.OnDismissListener) {
        dialog.setOnDismissListener(listener)
    }

    fun dismiss() {
        dialog.dismiss()
    }

}
